pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_events.go

```diff
@@ -252,9 +260,68 @@ func (d duplicateEventsEvaluator) testDuplicatedEvents(testName string, flakeOnl
 				continue
 			}
 		}
+		if strings.Contains(event.Message, "reason/TopologyAwareHintsDisabled") {
+			// Build the allowed time range only once
+			if buildTopologyHintAllowedTimeRange {
+				taintManagerTestIntervals := events.Filter(func(eventInterval monitorapi.Interval) bool {
+					return eventInterval.Source == monitorapi.SourceE2ETest &&
+						strings.Contains(eventInterval.StructuredLocator.Keys[monitorapi.LocatorE2ETestKey], "NoExecuteTaintManager")
+				})
+				// Start the allowed time range from time range of the tests. But events lag behind the tests since the tests do not wait
+				// until all dns pods are properly scheduled and reach ready state. So we will need to expand the allowed time range after.
+				for _, test := range taintManagerTestIntervals {
+					topologyHintAllowedTimeRange = append(topologyHintAllowedTimeRange, &timeRange{from: test.From, to: test.To})
```

Comment:
Lets log the test name and time range calculated here.
logrus.WithField("from": from).WithField("to": to).Info("found time range for test: %s", testName)